sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/ExternalCatalogUtils.scala

```diff
@@ -123,6 +124,38 @@ object ExternalCatalogUtils {
     }
     escapePathName(col) + "=" + partitionString
   }
+
+  def prunePartitionsByFilter(
```

Comment:
Is this method used? I think that this PR includes code that are not related to fixing 64KB issue.
I will investigate all of changes later.